datanode/dehistory/initialise.go

```diff
@@ -55,23 +55,25 @@ func DatanodeFromDeHistory(parentCtx context.Context, cfg InitializationConfig,
 			return fmt.Errorf("failed to get most recent history segment from peers:%w", err)
 		}
 
-		if mostRecentHistorySegmentFromPeers == nil {
-			log.Infof("no most recent segment is available from peers")
+		if response == nil {
+			log.Infof("unable to get a most recent segment response from peers")
 			return ErrDeHistoryNotAvailable
 		}
 
-		log.Infof("got most recent history segment:%s", mostRecentHistorySegmentFromPeers)
+		mostRecentHistorySegment := response.Response.Segment
 
-		toSegmentID = mostRecentHistorySegmentFromPeers.HistorySegmentId
+		log.Infof("got most recent history segment:%s from peer:%s", mostRecentHistorySegment, response.PeerAddr)
```

Comment:
Can we try to just us log fields instead of fmt directives?

Reply:
sorted